operator/src/main/java/org/keycloak/operator/Constants.java

```diff
@@ -56,8 +56,11 @@ public final class Constants {
     public static final String KEYCLOAK_HTTP_PORT_NAME = "http";
     public static final String KEYCLOAK_HTTPS_PORT_NAME = "https";
     public static final String KEYCLOAK_SERVICE_PROTOCOL = "TCP";
+    public static final String KEYCLOAK_HTTP_SERVICE_APP_PROTOCOL = "http";
```

Comment:
@vmuzikar When I changed to http2, I got an issue with the HTTP header size same as https://github.com/keycloak/keycloak/issues/13933. 
So should we keep it as "http" or make this value configurable from CRDs?

Reply:
> http2 is not an allowed value

It appears to be there: https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?search=https

> When I changed to http2, I got an issue with the HTTP header size same as https://github.com/keycloak/keycloak/issues/13933

Something similar was mentioned here https://github.com/keycloak/keycloak/discussions/10236 and one user ended up disabling http2 via a quarkus option. Did you also disable http2 for your keycloak server? Or are you saying that the appProtocol of http works just fine with http/2

It does seem like we should be able to figure out if there's some other quarkus option that needs to be tweaked so that http2 can continue to be used. For example we recently increased quarkus.http.limits.max-form-attribute-size in the keycloak defaults.

> So should we keep it as "http" or make this value configurable from CRDs?

If we don't want to account for quarkus.http.http2=false, then it seems like it would need to be configurable. We've talked about in the past for a way for the operator to interrogate the image or a running server to see what the effective state of the configuration is to make these kind of decisions, but we're a long way off from doing that.

Reply:
> Yes, the appProtocol = "http" works fine with http2 in my environment.

It seems like we may need the istio folks to weigh in here. I can't quite tell from their docs / issues, but it seems like not specifying http2 may mean that it will expect to downgrade to http based upon this setting. Or is that what you are expecting?

Also the istio docs indicate it can get the protocol from:

```
By the name of the port: name: <protocol>[-<suffix>].
```

Wouldn't that already be the case here, since we've named the port http?

Reply:
> I can't quite tell from their docs / issues, but it seems like not specifying http2 may mean that it will expect to downgrade to http based upon this setting. Or is that what you are expecting?

In my understanding, The Istio is a proxy for requests, and the http2 is the http1.1 improvement performance. 
So in my case, "http" is enough

> Wouldn't that already be the case here, since we've named the port http?

The Keycloak service is ok, but the Keycloak discovery service is missing a port name.

Reply:
> The Keycloak service is ok, but the Keycloak discovery service is missing a port name.

Let's just update this pr to do that instead, then we can side-step usage of the appProtocol for now.